solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -61,18 +70,179 @@ def run(site, model, init_time, start, end):
+    resampled = list(map(resampler, (*forecast, ac_power)))
     return resampled
+
+
+def run_persistence(observation, forecast, issue_time, index=False):
+    """
+    Run a persistence *forecast* for an *observation*.
+
+    For intraday forecasts, the *index* argument controls if the
+    forecast is constructed using persistence of the measured values
+    (*index = False*) or persistence using clear sky index or AC power
+    index.
+
+    For day ahead forecasts, only persistence of measured values
+    (*index = False*) is supported.
+
+    The persistence *window* is the time over which the persistence
+    quantity (irradiance, power, clear sky index, or power index) is
+    averaged over. The persistence window is automatically determined
+    based on the type of persistence forecast desired:
+
+      * Intraday persistence forecasts: Maximum of observation interval
+        length, forecast interval length, and forecast lead time to
+        start. No longer than 1 hour.
+      * Day ahead forecasts: Equal to forecast interval length.
+
+    Users that would like more flexibility may use the lower-level
+    functions in
+    :py:mod:`solarforecastarbiter.reference_forecasts.persistence`.
+
+    Parameters
+    ----------
+    observation : datamodel.Observation
+        The metadata of the observation to be used to create the
+        forecast.
+    forecast : datamodel.Forecast
+        The metadata of the desired forecast.
+    issue_time : pd.Timestamp
+        Issue time of the forecast run.
+    index : bool, default False
+        If False, use persistence of observed value. If True, use
+        persistence of clear sky or AC power index.
+
+    Returns
+    -------
+    forecast : pd.Series
+        Forecast conforms to the metadata specified by the *forecast*
+        argument.
+
+    Raises
+    ------
+    ValueError
+        If forecast and issue_time are incompatible.
+    ValueError
+        If forecast.run_length >= 1 day and index=True.
+    ValueError
+        If instantaneous forecast and instantaneous observation interval
+        lengths do not match.
+    ValueError
+        If average observations are used to make instantaneous forecast.
+    """
+    forecast_start, forecast_end = get_forecast_start_end(forecast, issue_time)
+
+    # intra day persistence and "day ahead" persistence require
+    # fairly different parameters.
+    # is this a sufficiently robust way to distinguish?
+    intraday = forecast.run_length < pd.Timedelta('1d')
+    if intraday:
+        # time window over which observation data will be used to create
+        # persistence forecast.
+        # assumes observation.interval_length <= 1h
+        window = min(
+            pd.Timedelta('1h'),
+            max(observation.interval_length, forecast.interval_length,
+                forecast.lead_time_to_start, forecast.run_length)
+        )
+        data_end = issue_time
```

Comment:
Implicit here is that the data continues up to `issue_time`.  Is that consistent with the meaning of `issue_time`?  I'm thinking that `issue_time` is set ahead of now to allow for processing, etc., to ensure that the 9:00am forecast is done by 9:00am.  If that's the case, then what happens if the forecast processing starts at 8:30am?

Reply:
Yes, that is implicit in the definition but I don't know what we'd do differently without further convolution of the logic. I think this is consistent with how persistence forecasts are used as a benchmark in research. I think the operational implications are beyond what we can do here. 

For calculation implications... in `persistence_scalar_index` I'm making an assumption that the `load_data` function returns `NaN` for any times that do not yet exist.

Reply:
May want to extend the window to encompass a minimum period of data.

Reply:
One option is to add a `run_time` parameter and set `data_end = run_time`. We'd still need to keep the `issue_time` parameter to determine the start/end times from the `forecast` attributes.

I don't want these functions to check the database for the most recent data and extend the window if necessary. I'd rather make a NaN forecast.

Reply:
I like the `run_time` idea.  Agree that a dynamic data window is not a good idea here.